src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double firstLeg;
+    private double secondLeg;
+
+    public RightTriangle() {
+    }
+
+    public RightTriangle(double firstLeg, double secondLeg) {
+        this.typeOfFigure = TypeOfFigure.RIGHT_TRIANGLE;
+        this.area = getTriangleArea(firstLeg, secondLeg);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead of calculating the area inside the constructor, consider using a separate method that can be called after the object is constructed. This will improve code readability.